docs/Validation-and-Serialization.md

```diff
@@ -502,6 +502,23 @@ fastify.post('/', { schema, attachValidation: true }, function (req, reply) {
 })
 ```
 
+If you want to format AJV errors yourself, you can provide a sync function that must return a string as the `errorFormatter` option to AJV when instantiating Fastify.
+
+`errors` in an array of AJV schema errors `ajv.ErrorObject`
+`dataVar` is the currently validated part of the scheme. (params | body | querystring | headers)
+```js
+const fastify = Fastify({
+  ajv: {
+      customOptions: {
+        errorFormatter: (errors, dataVar) => {
```

Comment:
I would prefer if this options to be moved _outside_ of the `ajv` namespace in the options. It's Fastify responsiblity to handle this, not ajv.

Reply:
I agree, but won't it be directly coupled anyways to `ajv` because of the `errors` type (which is an `ajv` type)

Reply:
I think this can also be a generic `Error` if we set the validator compiler to somethignelse with [`setValidatorCompiler`](https://www.fastify.io/docs/latest/Server/#setvalidatorcompiler), so the type is not really correct.

Reply:
So we want to override the default `validationCompiler` and implement one that complies the schemas and if there are errors it returns them as a custom fastify error / `Error`?

Reply:
That should be possible, yes.

Reply:
@mcollina How should the `Error` object be populated? should we only keep the `message` property? are there other fields we should use to construct the error message?

Reply:
@mcollina I have changed the types to support a custom fastify object that will contain the errors, `ajv` implements this interface by default, is this fine with you?